hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -263,6 +264,43 @@ public void updateClusterMetrics(ClusterMetrics st) {
     }
   }
 
+  private void updateBalancerTableLoadInfo(TableName tableName, Map<ServerName, List<RegionInfo>> loadOfOneTable) {
+    RegionHDFSBlockLocationFinder finder = null;
+    if ((this.localityCost != null) || (this.rackLocalityCost != null)) {
+      finder = this.regionFinder;
+    }
+    BalancerClusterState cluster =
+      new BalancerClusterState(loadOfOneTable, loads, finder, rackManager);
+
+    initCosts(cluster);
+    double total = 0;
+    double[] functionCosts = new double[costFunctions.size()];
+    for (int i = 0; i < costFunctions.size(); i++) {
+      CostFunction c = costFunctions.get(i);
+      functionCosts[i] = 0.0;
+      if (!c.isNeeded()) {
+        continue;
+      }
+      Float multiplier = c.getMultiplier();
+      double cost = c.cost();
+      functionCosts[i] = multiplier * cost;
+      total += functionCosts[i];
+    }
```

Comment:
this is largely re-implementing the `computeCosts` function. Can you re-use or unify with that function instead?